components/input.js

```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+import styled from 'styled-components/native'
+
+const TextBox = styled.TextInput`
+  border-bottom-width: 1px;
+  background-color: #fff;
+  border-Color: ${props => props.colorB ? props.colorB :'rgb(131,131,131)'};
+  padding-left: 5px;
+  margin: 10px;
+`
+const StyledInput = (props) => (
+    <TextBox
+      colorB = {props.colorB}
+      value = {props.value}
+      keyboardType = {props.type}
+      secureTextEntry = {props.secure}
+      placeholder = {props.label}
+      onChangeText={(text) => props.change ? props.change(text): ''}
```

Comment:
En lugar de utilizar un operador ternario `?`, podrias utilizar doble AND `&&`(no recuerdo su nombre), esto con la finalidad de solo tener el `if` y no el `else`, ejemplo:
*original*
`onChangeText={(text) => props.change ? props.change(text): ''}`

*Refactorización*
`onChangeText={(text) => props.change && props.change(text)}`


Ya que de no cumplirse la condición no queremos que haga algo, podemos utilizar el ejemplo de arriba.